Makefile

```diff
@@ -287,6 +287,10 @@ ifeq ($(ORIG_COMPILER),1)
   CCAS     := $(CC)
 endif
 
+# EGCS Compiler
+CC_EGCS := env COMPILER_PATH=tools/egcs/$(DETECTED_OS) tools/egcs/$(DETECTED_OS)/gcc
+CCAS_EGCS := $(CC_EGCS) -x assembler-with-cpp
```

Comment:
alright I guess we have conflicts for:

* `EGCS` prefix vs suffix
* -B vs `env COMPILER_PATH`

do you have opinions on either of these? I probably prefer prefix (because "EGCS_CFLAGS" reads more how I'd say it in English) but do not care too much about -B

Reply:
prefix is ok with me, I can change it

Do you know if there's any substantial differences between -B and env? Other projects use env so I went with that initially, but if there's no real difference between the two -B is probably cleaner

Reply:
After some light research:

* `-B` would also be used for libraries and headers search, which I think is morally correct though it wouldn't affect anything here
* `-B` also allows you to specific a prefix (e.g. `-B tools/egcs/macos/mips-linux-`) although that seems a bit weird if you also want libraries and headers, and again it wouldn't affect anything here